### ISSUE TITLE
App won't start with invalid plugins

### ISSUE THREAD

**Issue by jaimeiniesta** [sorentwo/oban#852]
### Environment

- Oban v2.14.2, Oban.Web v2.9.6, Oban.Pro v0.13.1 
- PostgreSQL 14
- Erlang/OTP 23 [erts-11.2.2.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1]
- Elixir 1.13.4 (compiled with Erlang/OTP 23)

### Current Behavior

I saw a warning about `Oban.Pro.Plugins.Reprioritizer` being deprecated so I changed it to `Oban.Pro.Plugins.DynamicReprioritizer` - then the app didn't start.

The problem was a typo, the correct name of the plugin is `Oban.Pro.Plugins.DynamicPrioritizer`.

In short, if you include an invalid module name like `Whatever` in the plugins, the app won't start and no warning will be issued.

### Expected Behavior

It would be nice to see a warning about that when starting the server if a plugin can't be started.

**Comment by sorentwo** [comment]
Oban will validate queues and plugins, including plugin options, but the use of `plugins: false` in the test environment hides it. By switching to `testing: :manual` for tests you’ll get validation errors for  queue/plugin options in test mode.

There are details about validation in tests in the [v2.12 CHANGELOG](https://hexdocs.pm/oban/2.12.0/changelog.html#testing-modes) and the [Testing Config guide](https://hexdocs.pm/oban/2.14.2/testing_config.html).

(This is the same fix as the staging errors issue in #853)

**Comment by jaimeiniesta** [comment]
Thanks, but I mean the `dev` environment.

No big deal, but the app won't start with invalid plugins in development.

**Comment by sorentwo** [comment]
I see what you mean. Having the testing mode should help catch issues before going to production at least. Keep in mind, adding unknown workers/servers to your application supervisor would crash without a warning as well.

**Comment by jaimeiniesta** [comment]
Hi, I was upgrading to 1.0.0 and saw in https://getoban.pro/docs/pro/1.0.0/changelog.html#v1-0-0-2023-06-30 this:

```
[Oban.Pro.Plugins.Reprioritizer] Renamed to Oban.Pro.Plugins.DynamicReprioritizer.
```

The correct name is `Oban.Pro.Plugins.DynamicPrioritizer`, right?

**Comment by sorentwo** [comment]
That's correct. I'll update the docs (though the deprecation warnings in prior versions are accurate)
